velox/exec/DistinctAggregations.cpp

```diff
@@ -123,10 +123,12 @@ class TypedDistinctAggregations : public DistinctAggregations {
         }
 
         rows.resize(data->size());
-        std::vector<VectorPtr> inputForAggregation_ =
-            makeInputForAggregation(data);
-        aggregate.function->addSingleGroupRawInput(
-            group, rows, inputForAggregation_, false);
+        if (rows.hasSelections()) {
```

Comment:
nit: `if (data->size() > 0)` before rows.resize(data->size()) to save some unnecessary resize operation.

Reply:
+1

```
  if (!data->empty()) {
        rows.resize(data->size());
        std::vector<VectorPtr> inputForAggregation_ =
            makeInputForAggregation(data);
        aggregate.function->addSingleGroupRawInput(
            group, rows, inputForAggregation_, false);
  }
```